include/swift/AST/Types.h

```diff
@@ -2993,11 +3039,13 @@ class AnyFunctionType : public TypeBase {
                   Type Output, RecursiveTypeProperties properties,
                   unsigned NumParams, ExtInfo Info)
   : TypeBase(Kind, CanTypeContext, properties), Output(Output) {
-    Bits.AnyFunctionType.ExtInfo = Info.Bits;
+    Bits.AnyFunctionType.ExtInfoBits = Info.Bits;
+    Bits.AnyFunctionType.HasUncommonInfo = Info.getUncommonInfo().hasValue();
```

Comment:
Please leave a comment saying that subclasses are responsible for initializing their uncommon-info storage.